src/mame/skeleton/digijet.cpp

```diff
@@ -2,71 +2,249 @@
+    ROM: ST M2764AF6
+    ADC0809CCN
+    LM2901
+    SN74LS373N
+    SN74LS00N
+    XTAL 7.372 MHz
+
+    _________________________
+    |          7   A 0       |
+    |          4   D 8       |
+    C          3   C 0    7  |
+    O          7     9    4  |
+    N  L2      3          0  |
+    N  M0    M2764AF6  X  0  |
+    |  91          80A39HL   |
+    |________________________|
+
+    Connector
+    1  Engine RPM
+    2  Coolant temperture (ADC IN2)
+    3  GND
+    4  Throttle switch (T0)
+    5  Lambda sensor
+    6  GND Air sensor
+    7  GND
+    8  Check engine (not populated) (P1.3)
+    9  GND
+    10 GND
+    11 Injector (P1.5)
+    12 Injector (P1.5)
+    13 Ignition (ADC IN1)
+    14 Air in temperature (ADC IN3)
+    15 Air amount (ADC IN0)
+    16 GND
+    17 GND
+    18 GND
+    19 Air sensor power
+    20 Fuel pump (P1.4)
+    21 Fuel pump (P1.4)
+    22 GND
+    23 Injector (P1.5)
+    24 Injector (P1.5)
+    25 GND
 
 **************************************************************************/
 
 /*
     TODO:
 
-    - Everything
+    - Figure out how the ADC is accessed
 */
 
 #include "emu.h"
 #include "cpu/mcs48/mcs48.h"
+#include "machine/timer.h"
+#include "machine/watchdog.h"
 
+#include "vw.lh"
 
 namespace {
 
-#define I8049_TAG   "i8049"
-
 class digijet_state : public driver_device
 {
 public:
 	digijet_state(const machine_config &mconfig, device_type type, const char *tag)
 		: driver_device(mconfig, type, tag)
-		, m_maincpu(*this, I8049_TAG)
+		, m_maincpu(*this, "maincpu")
+		, m_watchdog(*this, "watchdog")
+		, m_rpm_timer(*this, "rpm")
+		, m_io_adc(*this, "ADC%u", 0U)
 	{
 	}
 
 	void digijet(machine_config &config);
+	void digijet90(machine_config &config);
 
 private:
-	required_device<cpu_device> m_maincpu;
+	required_device<mcs48_cpu_device> m_maincpu;
+	required_device<watchdog_timer_device> m_watchdog;
+	required_device<timer_device> m_rpm_timer;
+	required_ioport_array<4> m_io_adc;
 
-	virtual void machine_start() override { }
+	virtual void machine_start() override ATTR_COLD;
 	virtual void machine_reset() override { }
 	void io_map(address_map &map) ATTR_COLD;
+	void prg_map(address_map &map) ATTR_COLD;
+
+	void p1_w(uint8_t data);
+
+	TIMER_DEVICE_CALLBACK_MEMBER( rpm_int );
+	uint8_t read_adc(offs_t offset);
+	void start_adc(uint8_t data);
+
+	bool m_rpm = false;
+	bool m_interrupt_enable = true;
+};
+
+void digijet_state::machine_start()
+{
+	save_item(NAME(m_rpm));
+	save_item(NAME(m_interrupt_enable));
 };
 
 void digijet_state::io_map(address_map &map)
 {
+	map(0x30, 0x3f).r(FUNC(digijet_state::read_adc));
+	map(0x30, 0x3f).w(FUNC(digijet_state::start_adc));
+};
+
+void digijet_state::prg_map(address_map &map)
+{
+	map(0x000, 0x7ff).rom().region("maincpu", 0);
+};
+
+uint8_t digijet_state::read_adc(offs_t offset)
+{
+	return m_io_adc[offset & 0x03]->read(); //FIXME this is a hack
+	// The address lines connected to the ADC are A8-A11 but those can't be accessed externally by this CPU
```

Comment:
P20-P23 outputs are time-multiplexed with A8-A11 on MCS-48 parts executing from external memory.

I thought that the ADC hookup might use the PROG expander protocol, but from other comments that doesn't sound like the case here.